src/PackagePublisher/Program.cs

```diff
@@ -20,13 +20,12 @@ public static class Program
         private const string ApiKeyEnvironmentVariableName = "APIKey";
         private static CommandOption _packagesDirectory;
         private static CommandOption _feedToUploadTo;
-        private static CommandOption _maxRetryCountOption;
 
         // Do NOT log the key
         private static string _apiKey = null;
         private static int _maxRetryCount = 5;
-        private static int _packagePushTimeoutInSeconds = 90;
-        private static int _semaphoreTimeoutInSeconds = _maxRetryCount * _packagePushTimeoutInSeconds;
+        private static readonly TimeSpan _packagePushTimeout = TimeSpan.FromSeconds(90);
+        private static readonly TimeSpan _semaphoreTimeout = TimeSpan.FromHours(1);
```

Comment:
I think this is fine for now, let's try and see what happens.  Another approach would be to keep a queue of packages to be uploaded, then start 4 tasks which each pull a package from the queue, upload in a retry loop, and repeat until the queue is empty.  This avoids the need for a semaphore.